src/ios.ts

```diff
@@ -132,11 +133,23 @@ export const iosApi = <RunTarget extends SupportedRunTarget<'ios'>>(
     target: { platform: 'ios', runTarget: options.runTarget },
     _internal: {
         ssh: async (...args) => {
+            let killProxyProcesss: (() => boolean) | undefined = undefined;
+
+            if (!options.targetOptions?.ip)
+                killProxyProcesss = await startUsbmuxProxy(22161, options.targetOptions?.port || 22);
+
             const ssh = await new NodeSSH().connect({
-                host: options.targetOptions!.ip,
+                host: options.targetOptions?.ip || '127.0.0.1',
+                port: options.targetOptions?.port ?? (options.targetOptions?.ip ? 22 : 22161),
                 username: 'root',
-                password: options.targetOptions!.rootPw || 'alpine',
+                password: options.targetOptions?.rootPw || 'alpine',
             });
+            ssh.connection?.on('error', (err) => {
+                if (err.level === 'client-socket') killProxyProcesss?.();
```

Comment:
Don't we want to always kill the proxy here? I haven't tested this, but I'm assuming that you would otherwise have to manually kill the proxy if you have an error (since you never check whether there's an existing proxy running but always try to start one on the same port).

Reply:
Well, I thought we always kill the proxy with each connection anyway (because we also throw away the connection). And then we need to clean up after ourselves in the case of an error that is related to the connection.

Reply:
> because we also throw away the connection

Does that actually happen when you rethrow the error here, though?

Reply:
I don’t know what you mean? If I reach that state, the connection already died.

Reply:
Huh? Why do you need the `killProxyProcess()` in the line I'm commenting on, then? o.o

Reply:
Well, the SSH connection died, maybe because the ssh server restarted for example. But because we open a new proxy for every connection, we still need to kill the proxy. It didn’t die. This is just a `finally` statement to clean up the proxy connections.

Reply:
I tried confirming my suspicion but was not able to trigger the `ssh.connection?.on('error')` callback. Uninstalling `openssh-server` while running `await ios._internal.ssh(['sleep', '10']);` causes the error below but that is _not_ thrown by the callback:

```
Error: read ECONNRESET
    at __node_internal_captureLargerStackTrace (node:internal/errors:490:5)
    at __node_internal_errnoException (node:internal/errors:620:12)
    at TCP.onStreamRead (node:internal/stream_base_commons:217:20) {
  errno: -104,
  code: 'ECONNRESET',
  syscall: 'read',
  level: 'client-socket'
}
```

At least port 22161 is not open anymore afterwards, though I'm pretty sure that's not because of either of the callbacks but rather because the Node process exited uncleanly.

Reply:
Since neither of us really seems to know what the error callback does: Is there any reason not to always kill the proxy, just in case?

```ts
export interface ClientErrorExtensions {
    /** Indicates 'client-socket' for socket-level errors and 'client-ssh' for SSH disconnection messages. */
    level?: string;
    /** Additional detail for 'client-ssh' messages. */
    description?: string;
}
```

Why wouldn't I want to do that in the case of "SSH disconnection"? And if the proxy was already killed elsewhere, who cares?